Misc/NEWS.d/next/Library/2024-07-13-13-25-31.gh-issue-121676.KDLS11.rst

```diff
@@ -0,0 +1,3 @@
+Now calling the Python implementation of the :meth:`functools.reduce` with a
+:term:`keyword argument` is raises a :exc:`DeprecationWarning`.
+This will be forbidden in Python 3.16 in order to match the C implementation.
```

Comment:
```suggestion
Deprecate calling the Python implementation of the :meth:`functools.reduce`
with a :term:`keyword argument`. This will be forbidden in Python 3.16 in
order to match the C implementation.
```

I think we should add a true What's News entry but I'm not sure if it's really needed since we are deprecating the Python implementation which is, in general, not used by default (unless it's a non-CPython implementation I think?)

Reply:
I think that adding a true What's New entry is worth adding because of non-CPython implementations that use the CPython stdlib.

Btw, I won't commit this suggestion because I am planning to rework this PR (see https://github.com/python/cpython/pull/121677#issuecomment-2445048029). After the rework your review will be is more than welcome! :slightly_smiling_face: